src/transaction/components/TransactionStatusLabel.tsx

```diff
@@ -0,0 +1,37 @@
+import { useMemo } from 'react';
+import { cn, color, text } from '../../styles/theme';
+import { useTransactionContext } from './TransactionProvider';
+import type { TransactionStatusLabelReact } from '../types';
+
+export function TransactionStatusLabel({
+  className,
+}: TransactionStatusLabelReact) {
+  const { errorMessage, isLoading, transactionId } = useTransactionContext();
+
+  // TODO: rethink ordering and add additional cases
+  const label = useMemo(() => {
```

Comment:
I think we can create an utility function, similar to [getSwapMessage](https://github.com/coinbase/onchainkit/blob/main/src/swap/core/getSwapMessage.ts), to help organize those cases.